main.go

```diff
@@ -140,178 +194,327 @@ func main() {
+	}
+
 	envsToUnset := []string{"GEM_HOME", "GEM_PATH", "RUBYLIB", "RUBYOPT", "BUNDLE_BIN_PATH", "_ORIGINAL_GEM_PATH", "BUNDLE_GEMFILE"}
 	for _, key := range envsToUnset {
 		if err := os.Unsetenv(key); err != nil {
 			fail("Failed to unset (%s), error: %s", key, err)
 		}
 	}
 
-	if configs.UseLegacyExport == "yes" {
-		log.Info("Using legacy export method...")
+	archive, err := xcarchive.NewMacosArchive(configs.ArchivePath)
+	if err != nil {
+		fail("Failed to parse archive, error: %s", err)
+	}
 
-		provisioningProfileName := ""
-		if configs.LegacyExportProvisioningProfileName != "" {
-			log.Detail("Using provisioning profile: %s", configs.LegacyExportProvisioningProfileName)
-			provisioningProfileName = configs.LegacyExportProvisioningProfileName
-		} else {
-			log.Detail("No provisining profile specified, let xcodebuild to grab one...")
-		}
+	// do a simple export if method set to none
+	{
+		if configs.ExportMethod == "none" {
+			log.Infof("Exporting app without re-sign...")
+
+			if err := utils.ExportAppFromArchive(configs.ArchivePath, configs.DeployDir, bitriseAppPathEnvKey); err != nil {
+				fail("Failed to archive app, error: %s", err)
+			}
 
-		exportingApp := true
-		if configs.LegacyExportOutputFormat == "pkg" {
-			exportingApp = false
+			log.Donef("The app path is now available in the Environment Variable: %s", bitriseAppPathEnvKey)
+			return
 		}
+	}
 
-		legacyExportCmd := xcodebuild.NewLegacyExportCommand()
-		legacyExportCmd.SetExportFormat(configs.LegacyExportOutputFormat)
+	exportMethod, err := exportoptions.ParseMethod(configs.ExportMethod)
+	if err != nil {
+		fail("Failed to parse export options, error: %s", err)
+	}
 
-		if exportingApp {
-			legacyExportCmd.SetExportPath(appPath)
-		} else {
-			legacyExportCmd.SetExportPath(pkgPath)
-		}
+	// legacy export
+	{
+		if configs.UseLegacyExport == "yes" {
+			log.Infof("Using legacy export method...")
 
-		legacyExportCmd.SetArchivePath(configs.ArchivePath)
-		legacyExportCmd.SetExportProvisioningProfileName(provisioningProfileName)
+			if xcodebuildVersion.MajorVersion >= 9 {
+				fail("Legacy export method (using '-exportFormat ipa' flag) is not supported from Xcode version 9")
+			}
 
-		log.Done("$ %s", legacyExportCmd.PrintableCmd())
-		fmt.Println()
+			provisioningProfileName := ""
+			if configs.LegacyExportProvisioningProfileName != "" {
+				log.Printf("Using provisioning profile: %s", configs.LegacyExportProvisioningProfileName)
 
-		if err := legacyExportCmd.Run(); err != nil {
-			fail("Export failed, error: %s", err)
-		}
+				provisioningProfileName = configs.LegacyExportProvisioningProfileName
+			} else {
+				log.Printf("Using embedded provisioning profile")
 
-		if exportingApp {
-			if err := utils.ExportOutputFile(appPath, appPath, bitriseAppPathEnvKey); err != nil {
-				fail("Failed to export %s, error: %s", bitriseAppPathEnvKey, err)
+				provisioningProfileName = archive.Application.ProvisioningProfile.Name
```

Comment:
archive.Application.ProvisioningProfile mac os archive's profile is mostly nil, check it and fail if no profile embedded!